src/hash.ts

```diff
@@ -1,9 +1,26 @@
 /** @module ssz */
 import { sha256 } from "js-sha256";
+import { cache } from "./cache";
+
+/** @ignore */
+function _hash(input: Buffer): Buffer {
+  return Buffer.from(sha256.update(input).arrayBuffer());
+}
 
 /**
  * Hash used for hashTreeRoot
  */
-export function hash(...inputs: Buffer[]): Buffer {
-  return Buffer.from(inputs.reduce((acc, i) => acc.update(i), sha256.create()).arrayBuffer());
+export function hash(input: Buffer): Buffer {
+  if (cache) {
+    const hexInput = input.toString('hex');
```

Comment:
I think it would be better for the sake of memory to do some fast hash like https://www.npmjs.com/package/farmhash of input and use it as key